exchanges/ftx/ftx_wrapper.go

```diff
@@ -1348,104 +1365,233 @@ func (f *FTX) ScaleCollateral(ctx context.Context, subAccount string, calc *orde
+			return nil, fmt.Errorf("%v %v %v calculateTotalCollateralOnline %w", f.Name, calc.CollateralAssets[x].Asset, calc.CollateralAssets[x].CollateralCurrency, order.ErrOfflineCalculationSet)
 		}
+	wallets:
+		for y := range wallet {
+			if !currency.NewCode(wallet[y].ID).Match(calc.CollateralAssets[x].CollateralCurrency) {
+				continue
+			}
+			for z := range balances {
+				if !currency.NewCode(balances[z].Coin).Match(calc.CollateralAssets[x].CollateralCurrency) {
+					continue
+				}
+				if len(pos) > 0 {
+					// use pos unrealisedPNL, not calc.collateralAssets'
+					calc.CollateralAssets[x].UnrealisedPNL = decimal.Zero
+					for i := range pos {
+						var futurePair currency.Pair
+						futurePair, err = currency.NewPairFromString(pos[i].Future)
+						if err != nil {
+							return nil, err
+						}
+						if !futurePair.Base.Match(calc.CollateralAssets[x].CollateralCurrency) {
+							continue
+						}
+						calc.CollateralAssets[x].UnrealisedPNL = calc.CollateralAssets[x].UnrealisedPNL.Add(decimal.NewFromFloat(pos[i].UnrealizedPNL))
+					}
+				}
+				collateralWeight := decimal.NewFromFloat(wallet[y].CollateralWeight)
+				usdValue := decimal.NewFromFloat(balances[z].USDValue)
+				total := decimal.NewFromFloat(balances[z].Total)
+				if total.IsZero() {
+					result.BreakdownByCurrency = append(result.BreakdownByCurrency, order.CollateralByCurrency{
+						Currency: calc.CollateralAssets[x].CollateralCurrency,
+					})
+					continue
+				}
+				scaling := usdValue.Div(total).Mul(collateralWeight)
+				avail := decimal.NewFromFloat(balances[z].AvailableWithoutBorrow)
+				lockedS := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedInStakes)
+				lockedC := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedAsCollateral)
+				lockedF := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedInFeeVoucher)
+				lockedN := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedInNFTBids)
+				lockedO := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedInSpotOrders)
+				lockedFO := decimal.NewFromFloat(balances[z].LockedBreakdown.LockedInSpotMarginFundingOffers)
+				locked := lockedS.Add(lockedC.Add(lockedF.Add(lockedN.Add(lockedO.Add(lockedFO)))))
+
+				breakDown := order.CollateralByCurrency{
+					Currency:          calc.CollateralAssets[x].CollateralCurrency,
+					ScaledTotal:       total.Mul(scaling),
+					ScaledMaintenance: total.Mul(scaling),
+					ScaledFree:        avail.Mul(scaling),
+					ScaledTotalLocked: locked.Mul(scaling),
+					OriginalTotal:     calc.CollateralAssets[x].FreeCollateral.Add(calc.CollateralAssets[x].LockedCollateral),
+					UnrealisedPNL:     calc.CollateralAssets[x].UnrealisedPNL,
+				}
+
+				if !locked.IsZero() {
+					if result.LockedBreakdown == nil {
+						result.LockedBreakdown = &order.CollateralLockedBreakdown{}
+					}
+
+					breakDown.ScaledLockedBreakdown = &order.CollateralLockedBreakdown{
+						LockedInStakes:                  lockedS.Mul(scaling),
```

Comment:
these `lockedS.Mul(scaling)` and others can be taken out of this scope to be used below in results locked breakdown.